### ISSUE TITLE
import unresolved - CLI reports no errors

### ISSUE THREAD

**Issue by magus** [AtomLinter/linter-eslint#943]
## Issue Type
Bug

## Issue Description
Similar to #610, CLI reports no errors but linter-eslint is unable to resolve.

## Bug Checklist
- Restart Atom
- Verify the `eslint` CLI gives the proper result, while `linter-eslint` does not
- Paste the output of the `Linter Eslint: Debug` command from the Command Palette below

```
Atom version: 1.17.2
linter-eslint version: 8.2.1
ESLint version: 2.12.0
Hours since last Atom restart: 0.1
Platform: darwin
Using local project ESLint from: .../node_modules/eslint
Current file's scopes: [
  "source.js.jsx",
  "meta.class.body.js",
  "meta.function.method.js",
  "meta.brace.round.js"
]
linter-eslint configuration: {
  "lintHtmlFiles": false,
  "useGlobalEslint": false,
  "showRuleIdInMessage": true,
  "disableWhenNoEslintConfig": true,
  "eslintrcPath": "",
  "globalNodePath": "",
  "advancedLocalNodeModules": "",
  "eslintRulesDir": "",
  "disableEslintIgnore": false,
  "disableFSCache": false,
  "fixOnSave": false,
  "scopes": [
    "source.js",
    "source.jsx",
    "source.js.jsx",
    "source.babel",
    "source.js-semantic"
  ],
  "rulesToSilenceWhileTyping": [],
  "rulesToDisableWhileFixing": []
}
```

**Comment by Arcanemagus** [comment]
Without an example set of code and steps to reproduce this will be impossible to debug. As an example the last report of this turned out to be an invalid `webpack` configuration that was assuming things were running in a CLI, but the error saying what was wrong was being hidden by `eslint-plugin-import`.

**Comment by magus** [comment]
Making a minimal project and configuration it seems like having my webpack config files in a directory is causing the issue. The relevant parts of the `.eslintrc.js` file...


```js
"settings": {
    "import/resolver": {
      "webpack": {
        "config": "./webpack.config.js",
      },
    },
  },
```
![screen shot 2017-06-19 at 3 45 38 pm](https://user-images.githubusercontent.com/290084/27309021-8b4942f4-5506-11e7-9b13-697495af331a.png)


```
"settings": {
    "import/resolver": {
      "webpack": {
        "config": "./webpack/webpack.config.js",
      },
    },
  },
```
![screen shot 2017-06-19 at 3 45 11 pm](https://user-images.githubusercontent.com/290084/27309019-86e056bc-5506-11e7-8320-cb0b8e36438c.png)

**Comment by Arcanemagus** [comment]
@magus If you read my comment, you can see that the last one of these that I investigated the problem was in the `webpack.config.js` file. The code you posted above is unfortunately useless for tracking down what is wrong with your setup. As I stated, we would need a _full_ set of code in a test repository, zip, or gist of all the files in order to debug this.

**Comment by Arcanemagus** [comment]
@tvkhoa since you 👍 this, the same goes to you: In order to investigate this a full example project would be needed.

**Comment by magus** [comment]
[eslint-test.zip](https://github.com/AtomLinter/linter-eslint/files/1089302/eslint-test.zip)

**Comment by magus** [comment]
I tracked down the issue to an error being thrown in the require of the webpack config files. It looks like `window` is defined but the location has some unexpected values. Is there any way to debug the linter-eslint command so I can add some log statements?

**Comment by tvkhoa** [comment]
@Arcanemagus 
In attempt to reproduce it, I tried to copy the config to new project and it works as normal
Then, I cloned my project to another place and it still worked
--
Finally, I figured out the problem is because I put my project in a uppercase folder. I just rename it to lowercase and it works

**Comment by magus** [comment]
That was not the issue for me, my project folder is named 'eslint-test'. You can easily reproduce my issue with the provided zip above, it seems related to throwing errors.

**Comment by magus** [comment]
@Arcanemagus I've provided an example

**Comment by Arcanemagus** [comment]
@magus Sorry for the delay in looking into this.

In the time since you posted your screenshots it seems ESLint output has changed as now it _just_ shows `Resolve error: `.
![image](https://user-images.githubusercontent.com/427137/30532649-481fb1de-9c0a-11e7-9912-4440a6840015.png)

Whether due to updates to dependencies in the meantime, or the directory fixes in v8.3.0, I'm seeing the same result regardless of the location of the `webpack.config.js` now, unlike what you were seeing in https://github.com/AtomLinter/linter-eslint/issues/943#issuecomment-309594715.

----

As to why this isn't working, my initial example and your guess that this "sounds like it may be very similar" turned out to be spot on:

Your `throwError.js` script is throwing an error because `window` is defined, but it can't determine the host name (since `window.location` is `undefined` in the ESLint worker thread). This is complicated by the fact that `eslint-import-resolver-webpack` has a nice "feature" where errors thrown from configuration objects are silently ignored and never passed on, so it's impossible to see this until you start throwing debug log statements everywhere in the code to see where it's actually failing.

tl;dr? Make your `throwError.js` script more tolerant of environments where `window` is defined but can't give you a location 😉.
